packages/core/core/src/UncommittedAsset.js

```diff
@@ -122,13 +123,8 @@ export default class UncommittedAsset {
     this.value.committed = true;
   }
 
-  async getCode(): Promise<string> {
-    if (this.ast != null && this.isASTDirty) {
-      throw new Error(
-        'Cannot call getCode() on an asset with a dirty AST. For transformers, implement canReuseAST() and check asset.isASTDirty.',
-      );
-    }
-
+  // Please do not use this outside of uncomittedasset. Thanks :)
+  async __INTERNAL_getContent(): Promise<string> {
```

Comment:
You call this in the same class, so I would make factor this out into `function getAssetContent(asset: UncommitedAsset) {...}` and call that in `getCode` and `loadExistingSourcemap `

Reply:
That's a wonderful idea, am gonna do that right away. Was gonna figure out why tests fail anyway :)